README.md

```diff
@@ -15,218 +15,380 @@ and understand how to run and work with Dgraph.
+using var dgraphClient = DgraphClient.Create(channel);
 ```
 
 
 ### Altering the Database
 
-To set the schema, pass the schema into the `DgraphClient.Alter` function, as seen below:
+To set the schema, create an instance of `Dgraph.Api.Operation` and use the `Alter` endpoint.
 
 ```c#
-var schema = "name: string @index(exact) .";
-var result = client.Alter(new Operation{ Schema = schema });
+using Dgraph;
+
+var operation = new Api.Operation {
+    Schema = "name: string @index(exact) ."
+};
+var response = await dgraphClient.Alter(operation);
+if (response.IsFailed) {
+    // Handle errors
+}
 ```
+`Operation` contains other fields as well, including `DropAttr` and `DropAll`. `DropAll` is useful if you wish to discard all the data without bringing the instance down. `DropAttr` is used to drop all the data related to a predicate.
 
-The returned result object is based on the FluentResults library. You can check the status using `result.isSuccess` or `result.isFailed`. More information on the result object can be found [here](https://github.com/altmann/FluentResults).
+Starting in Dgraph version 20.03.0, indexes can be computed in the background. You can set the `RunInBackground` field to `true` like so:
 
+```c#
+using Dgraph;
+
+var operation = new Api.Operation {
+    Schema = "name: string @index(exact) .",
+    RunInBackground = true
+};
+var response = await dgraphClient.Alter(operation);
+if (response.IsFailed) {
+    // Handle errors
+}
+```
 
 ### Creating a Transaction
 
-To create a transaction, call `DgraphClient.NewTransaction` method, which returns a
-new `Transaction` object. This operation incurs no network overhead.
+To create a transaction, call the `IDgraphClient.NewTransaction()` method, which returns a new `ITransaction`. This operation incurs no network overhead.
 
-It is good practice to call to wrap the `Transaction` in a `using` block, so that the `Transaction.Dispose` function is called after running
-the transaction. 
+To ensure the `ITransaction` is properly disposed after it has completed, use the `using` keyword.
 
 ```c#
-using(var transaction = client.NewTransaction()) {
-    ...
+using var transaction = dgraphClient.NewTransaction();
+var transactionResponse = await transaction.Mutate(...);
+if (transactionResponse.IsFailed) {
+    // Handle errors
+}
+var response = await transaction.Commit();
+if (response.IsFailed) {
+    // Handle errors
 }
 ```
 
-You can also create Read-Only transactions. Read-Only transactions only allow querying, and can be created using `DgraphClient.NewReadOnlyTransaction`.
-
+Read-only transactions can be created by calling the `IDgraphClient.NewReadOnlyTransaction` method. Read-only transactions are useful to increase read speed because they can circumvent the usual consensus protocol. Read-only transactions cannot contain mutations. There is nothing to dispose for a `ReadOnlyTransaction` object, so it does not implement `IDisposable`.
```

Comment:
This is a very long line, we should limit the width of the line to 100/120 or so. We should do this throughout the file.